build.sbt

```diff
@@ -1,17 +1,61 @@
-name := "morganey"
 
-organization := "me.rexim"
+/*
+ * =================================== settings ===================================
+ */
 
-version := "0.0.1-SNAPSHOT"
+lazy val commonSettings = Seq(
+  name := "morganey",
+  organization := "me.rexim",
+  version := "0.0.1-SNAPSHOT",
+  scalaVersion := "2.11.7",
+  scalacOptions ++= Seq("-unchecked", "-deprecation", "-feature"),
+  libraryDependencies ++= Seq(
+    "jline" % "jline" % "2.6",
+    "org.scalatest" % "scalatest_2.11" % "2.2.4" % "test"
+  )
+)
 
-scalaVersion := "2.11.7"
+lazy val macroSettings = Seq(
+  name := "morganey-macros",
+  libraryDependencies += "org.scala-lang" % "scala-reflect" % scalaVersion.value
+)
 
-scalacOptions ++= Seq("-unchecked", "-deprecation", "-feature")
+lazy val kernelSettings = Seq(
+  name := "morganey-kernel",
+  libraryDependencies += "org.scala-lang.modules" %% "scala-parser-combinators" % "1.0.4"
+)
 
```

Comment:
I'm not an expert in macros, but why was it so essential to split the project into subprojects to be able to work with macros? Don't get me wrong I like the idea. :) `meta` works above `kernel` and `kernel` knows nothing about `meta`. This looks right to me. But still what was the motivation to split the project? Was this something related to the technical limitations of macros or just a project structure improvement?